### ISSUE TITLE
connect ECONNREFUSED 127.0.0.1:25 non-local smtp server

### ISSUE THREAD

**Issue by ernestomancebo** [jhuckaby/pixl-mail#1]
I get a

> connect ECONNREFUSED 127.0.0.1:25

error when trying to send an email under a scenario similar to this:

```

var smtp = 'my.smtp.com';
var mail = new PixlMail(smtp);
...
mail.send(msg, placeholder, function(err) {
   console.log(err);
});
```

I'm using node 4.3.1 and pixl-mail 1.0.5

**Comment by jhuckaby** [comment]
It sounds like you have `my.smtp.com` mapped to `127.0.0.1` in your local `/etc/hosts` file?  It certainly doesn't resolve to 127.0.0.1 if I try to look it up from here (I get `74.91.92.151`).

If this is the case, and you expect it to resolve to `127.0.0.1`, then it sounds like you don't have sendmail actually running on your local machine.  It is refusing connections to port 25.  You can test this by telnetting to it in your local terminal:

```
telnet 127.0.0.1 25
```

If you are on a Mac, I followed these instructions to get sendmail (postfix) running locally: http://blog.campodoro.org/?p=2318

Good luck!

\- Joe

**Comment by ernestomancebo** [comment]
Actually I'm using the `smtp.mailgun.org`, which is mapped to `209.61.151.224` and get the message that states that can't connect to 127.0.0.1:25.

By the way, it also happens on the other constructor, which requires the server address and the port.

I used the `my.smtp.com` as an foobar when reported the issue, my bad.

**Comment by jhuckaby** [comment]
Oh my, I am so sorry.  It looks like one of our dependencies, [nodemailer](https://www.npmjs.com/package/nodemailer), made a breaking API change recently, and changed the name of `hostname` to just `host` in their SMTP transport options object.

I will fix this in pixl-mail ASAP, but in the meantime, you can work around this problem by calling `setOption()` and passing in `host` as the key, and `smtp.mailgun.org` as the value, before calling `send()`.  Example:

``` js
var mail = new PixlMail();
...
mail.setOption( 'host', 'smtp.mailgun.org' ); // Add this here
...
mail.send(msg, placeholder, function(err) {
   console.log(err);
});
```

Thank you very much for catching this @ernestomancebo!  It'll be fixed very soon.  I'm leaving this issue open until it is.

\- Joe

**Comment by jhuckaby** [comment]
Fixed in v1.0.6: https://github.com/jhuckaby/pixl-mail/releases/tag/v1.0.6

Commit: https://github.com/jhuckaby/pixl-mail/commit/12bbf45dd5077a5ffab136078078e508ec0a7ac0

You should no longer need the workaround.

Thanks again!

**Comment by ernestomancebo** [comment]
Thanks to you.
